### ISSUE TITLE
Implement Expect

### ISSUE THREAD

**Issue by HEmile** [HEmile/storchastic#28]
Implementing Expect is not as easy as I hoped. Unfortunately, in discrete VAEs, for example, you pass logits with shape `[batch, latents, categories]`. `enumerate_support()` does not recognize that `batch` is independent, but the different `latents` are dependent! It only enumerates the support over the `categories` dimension, but it should also enumerate over the `latents` dimension. To do this automatically however requires independency assumptions, which probably brings us back to Issue #4.
